pkg/query/paging/response.go

```diff
@@ -1,22 +1,26 @@
 package paging
 
 // Response represents a response object containing information about pagination and total count of records.
-//   - PageIndex: The index of the page (starting from 0).
-//   - PageSize: The number of records per page.
+//   - PageIndex: The index of the page (starting from 0). This is required.
+//   - PageSize: The number of records per page. This is required.
+//   - TotalDisplayableResults: The total number of results that can be displayed. This is required.
+//   - TotalResults: The total number of results, including those that may not be displayed. This is optional and will be omitted if empty.
 type Response struct {
-	PageIndex     int    `json:"index"`
-	PageSize      int    `json:"size"`
-	TotalRowCount uint64 `json:"total"`
+	PageIndex               int    `json:"index" binding:"required"`
+	PageSize                int    `json:"size" binding:"required"`
+	TotalDisplayableResults uint64 `json:"totalDisplayableResults" binding:"required"`
+	TotalResults            uint64 `json:"totalResults,omitempty"`
 }
 
-func NewResponse(request *Request, totalRowCount uint64) *Response {
+func NewResponse(request *Request, totalDisplayableResults, totalResults uint64) *Response {
```

Comment:
The agreement with frontend was, `for backwards compatibility totalResults must never be set if totalResults == totalDisplayableResults`. So that a backend is not ever required to set `totalResults`.

I would suggest the following:

Keep the original function which can be used when there is no intention by the backend to distinguish between `TotalDisplayableResults` and `TotalResults`. So `NewResponse` will always only set `TotalDisplayableResults`.

And then you can add a new function which has both parameters `TotalDisplayableResults` and `TotalResults` and ensures `totalResults must never be set if totalResults == totalDisplayableResults`

Reply:
So how does this affect [NewMetadata](https://github.com/greenbone/opensight-golang-libraries/blob/0521c1896a08ab09ad385f974e34c8e027949df4/pkg/query/responseWithMetadata.go#L33) function? Do we add a new to handle `totalDisplayableResults`: (e.g `NewMetadataWithDisplayableResults()`)? 

Ref [caller example in VI backend](https://github.com/greenbone/vulnerability-intelligence-backend/blob/98c9163783c70ac129c2c275dcbddba1d1d9deac/pkg/web/vts/vts.go#L139)

Reply:
Yes that makes sense.

However `totalDisplayableResults` is always set, so the extra function should maybe be called `NewMetadataWithTotalResults`

Reply:
Thanks. 
Suggestion added